src/cloffeine/async_loading_cache.clj

```diff
@@ -1,48 +1,58 @@
 (ns cloffeine.async-loading-cache
   (:refer-clojure :exclude [get])
-  (:require [cloffeine.common :as common]
-            [cloffeine.async-cache :as acache]
+  (:require [cloffeine.async-cache :as acache]
+            [cloffeine.common :as common]
             [cloffeine.loading-cache :as loading-cache])
-  (:import [com.github.benmanes.caffeine.cache 
-            AsyncCache 
-            AsyncLoadingCache 
-            AsyncCacheLoader]))
+  (:import [com.github.benmanes.caffeine.cache
+            AsyncCache
+            AsyncLoadingCache
+            AsyncCacheLoader
+            CacheLoader]))
 
 (defn make-cache
   "Create an AsyncLoadingCache. See `cloffeine.common/builder` for settings.
-   A semi-persistent mapping from keys to values. Values are automatically loaded by the cache asynchronously, and are stored in the cache until either evicted or manually invalidated.)
-Implementations of this interface are expected to be thread-safe, and can be safely accessed by multiple concurrent threads."
-  (^AsyncLoadingCache [cache-loader]
+  A semi-persistent mapping from keys to values. Values are automatically loaded
+  by the cache synchronously, and are stored in the cache until either evicted
+  or manually invalidated.
+  Implementations of this interface are expected to be thread-safe, and can be
+  safely accessed by multiple concurrent threads."
+  (^AsyncLoadingCache [^CacheLoader cache-loader]
    (make-cache cache-loader {}))
-  (^AsyncLoadingCache [cache-loader settings]
+  (^AsyncLoadingCache [^CacheLoader cache-loader settings]
```

Comment:
was there a reflection warning?

Reply:
Yes, the runtime couldn't disambiguate between overloads.